File: hackerRank-js/algorithms/diagonal-difference.js
rl = require('readline');
var reader = rl.createInterface({
  input:process.stdin,
  output:process.stdout
});

function inputFunc(reader){
  let lines = new Array;
  let array = new Array;
  reader.on('line',function(line){
    lines.push(line);
  });
  reader.on('close',function(){
    var arr = new Array;
    var i = 0;
    var l = 0;
    var r = 0;
    var numrowcolumns = Number(lines[0])-1;
    var leftcount = 0;
    var rightcount = 0;
    while(i < Number(lines[0])){
      array[i] = lines[i+1].split(' ');
      arr[i] = array[i].map((a)=>Number(a));
      i++;
    }
    while(l < Number(lines[0])){
      leftcount += arr[l][l]
      l++;
    }
    while(r < Number(lines[0])){
      rightcount += arr[r][numrowcolumns]
      r ++;
      numrowcolumns --;
    }
    let count = leftcount - rightcount;
    let abscount = count < 0 ? Math.abs(count) : count
    console.log(abscount);
  });
}
inputFunc(reader);

File: hackerRank-rb/algorithms/diagonal-difference.rb
input = readlines
def diagonalDiff(input)
  i = 0
  l = 0
  r = input[0].to_i - 1
  leftcount = 0
  rightcount = 0
  array = []
  arr = []
  while i < input[0].to_i
    array[i] = input[i+1].split(' ') #array型を返す関数splitなのでto_iできない
    arr[i] = array[i].map{|a| a.to_i}
    i += 1
  end
  arr.each do |a|
    leftcount += a[l]
    l += 1
  end
  arr.each do |a|
    rightcount += a[r]
    r -= 1
  end
  count = leftcount - rightcount
  abscount = count < 0 ? count.abs : count
  p abscount
end
diagonalDiff(input)
